lib/cisco_node_utils/vlan.rb

```diff
@@ -100,19 +100,19 @@ def fabricpath_feature_set(fabricpath_set)
     def mode
       result = config_get('vlan', 'mode', @vlan_id)
       return default_mode if result.nil?
-      result.downcase! if result[/FABRICPATH/]
-      result
+      result[0].downcase! if result[0][/FABRICPATH/]
+      result[0]
     end
 
     def mode=(str)
-      if str.empty?
+      if str[0] == default_mode
         config_set('vlan', 'mode', @vlan_id, 'no', '')
       else
-        if 'fabricpath' == str
+        if 'fabricpath' == str[0]
           fabricpath_feature_set(:enabled) unless
             :enabled == fabricpath_feature
         end
-        config_set('vlan', 'mode', @vlan_id, '', str)
+        config_set('vlan', 'mode', @vlan_id, '', str[0])
       end
     end
```

Comment:
Hi. Yes you are right. The setter input should be a string. Changed the code to reflect this.
I missed that there was a minitest for vlan_mt_full. I tried this morning but I do not have a 7k with
a F3 LC. That seems to be the only LC where this tests can be run.